apps/coscrad-frontend/src/components/resources/factories/full-view-resource-presenter-factory.tsx

```diff
@@ -0,0 +1,44 @@
+import { ResourceType } from '@coscrad/api-interfaces';
+import { FunctionalComponent } from '../../../utils/types/functional-component';
+import { BibliographicReferenceDetailPresenter } from '../bibliographic-references/bibliographic-reference-detail.presenter';
+import { BookDetailPresenter } from '../books/book-detail.presenter';
+import { MediaItemDetailPresenter } from '../media-items/media-item-detail.presenter';
+import { PhotographDetailPresenter } from '../photographs/photograph-detail.presenter';
+import { SongDetailPresenter } from '../songs/song-detail.presenter';
+import { SpatialFeatureDetailPresenter } from '../spatial-features/spatial-feature-detail.presenter';
+import { TermDetailPresenter } from '../terms/term-detail.presenter';
+import { TranscribedAudioDetailPresenter } from '../transcribed-audio/transcribed-audio-detail.presenter';
+import { VocabularyListDetailPresenter } from '../vocabulary-lists/vocabulary-list-detail.presenter';
+import { withCommands } from './with-commands';
+
+/**
+ * TODO We could have a mapped type if we need type safety here.
+ */
+const lookupTable: { [K in ResourceType]: FunctionalComponent } = {
+    [ResourceType.bibliographicReference]: BibliographicReferenceDetailPresenter,
+    [ResourceType.book]: BookDetailPresenter,
+    [ResourceType.mediaItem]: MediaItemDetailPresenter,
+    [ResourceType.photograph]: PhotographDetailPresenter,
+    [ResourceType.song]: SongDetailPresenter,
+    [ResourceType.spatialFeature]: SpatialFeatureDetailPresenter,
+    [ResourceType.term]: TermDetailPresenter,
+    [ResourceType.transcribedAudio]: TranscribedAudioDetailPresenter,
+    [ResourceType.vocabularyList]: VocabularyListDetailPresenter,
+};
+
+/**
+ * This is a concrete Resource Presenter Factory
```

Comment:
Say more or remove the comment. You'll want a similar comment in the thumbnail factory as well in case you keep it.

Reply:
done here and for the thumbnail factory